itests/kit/node_unmanaged.go

```diff
@@ -440,66 +379,228 @@ func (tm *TestUnmanagedMiner) waitForMutableDeadline(ctx context.Context, sector
 	}
 }
 
-func (tm *TestUnmanagedMiner) OnboardCCSector(ctx context.Context, proofType abi.RegisteredSealProof) (abi.SectorNumber, chan WindowPostResp,
-	context.CancelFunc) {
-	req := require.New(tm.t)
+func (tm *TestUnmanagedMiner) NextSectorNumber() abi.SectorNumber {
 	sectorNumber := tm.currentSectorNum
 	tm.currentSectorNum++
+	return sectorNumber
+}
+
+func (tm *TestUnmanagedMiner) PrepareSectorForProveCommit(
+	ctx context.Context,
+	proofType abi.RegisteredSealProof,
+	sectorNumber abi.SectorNumber,
+	pieces []abi.PieceInfo,
+) (seedEpoch abi.ChainEpoch, proveCommit []byte) {
+
+	req := require.New(tm.t)
+
+	// Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
+	preCommitSealRandEpoch := tm.waitPreCommitSealRandomness(ctx, sectorNumber, proofType)
+
+	// Generate a Pre-Commit for the CC sector -> this persists the proof on the `TestUnmanagedMiner` Miner State
+	tm.generatePreCommit(ctx, sectorNumber, preCommitSealRandEpoch, proofType, pieces)
 
 	// --------------------Create pre-commit for the CC sector -> we'll just pre-commit `sector size` worth of 0s for this CC sector
 
-	// Step 1: Wait for the pre-commitseal randomness to be available (we can only draw seal randomness from tipsets that have already achieved finality)
-	preCommitSealRand := tm.waitPreCommitSealRandomness(ctx, sectorNumber)
+	if !proofType.IsNonInteractive() {
```

Comment:
What happens if it is NI ? Will that be addressed in https://github.com/filecoin-project/lotus/pull/12105 ?

Reply:
nope, nothing needed here for it - all we do is skip precommit and proceed as if we'd done it and it just works

Reply:
* interactive is: PC1, PC2, `<submit precommit>`, C1, C2, `<submit provecommit>`
* non-interactive is simply: PC1, PC2, C1, C2, `<submit provecommit>`